### PR TITLE
Remove deprecated function SparseLUDecomposition::decompose.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

File: include/deal.II/lac/sparse_decomposition.h [hunk 1]
```diff
@@ -237,24 +237,6 @@ class SparseLUDecomposition : protected SparseMatrix<number>,
...
-
-  /**
-   * This method is deprecated, and left for backward compability. It will be
-   * removed in later versions.
-   *
-   * @deprecated
-   */
-  virtual bool is_decomposed () const DEAL_II_DEPRECATED;
-
-  /**
```
Comment (kronbichler) [inline include/deal.II/lac/sparse_decomposition.h]: The documentation of the 'decomposed' flag (protected data member, line 313) still refers to decompose(). Could you please remove that reference and simply state that the derived classes will use this flag for tracking the state of the decomposition?

Comment (bangerth) [inline include/deal.II/lac/sparse_decomposition.h]: Good catch. It turns out that this flag is now unused altogether (only ever written to, never read), so I'm going to remove it. But there is also a big blob of documentation in `SparseLUDecomposition` that I will have to review.

**Comment by bangerth** [comment]
OK, flag removed, documentation updated.

**Comment by kronbichler** [comment]
Looks good, thanks.

**Pull request merged**